lib/installer_state_controller.dart

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import 'dart:async';
+
+enum InstallerState {
+  kInitializing,
+  kUnpacking,
+  kSettingUp,
+  kRunning,
+  kDone,
+  kError
```

Comment:
the k-prefix is generally used for constants :)

Reply:
I wanted to name the enum values with upper case initials, but it is not idiomatic so I cheated the linter 😅 
Fixed to the idiomatic way.